modules/redhat_manifest.py

```diff
@@ -0,0 +1,293 @@
+short_description: Interact with a Red Hat Satellite Subscription Manifest
+description:
+    - Download and modify a Red Hat Satellite Subscription Manifest
+author:
+    - "Sean O'Keeffe (@sean797)"
+options:
+    name:
+        description:
+           - Manifest Name
+    uuid:
+        description:
+           - Manifest uuid
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    pool_id:
+        description:
+            - Subscription pool_id
+    quantity:
+        description:
+            - quantity of pool_id Subscriptions
+        type: int
+    pool_state:
+        description:
+            - Subscription state
+        default: present
+        choices:
+            - present
+            - absent
+    state:
+        description:
+            - Manifest state
+        default: present
+        choices:
+            - present
+            - absent
+    path:
+        description:
+            - path to export the manifest
+        type: path
+    verify_ssl:
+        description:
+            - Validate Portal SSL
+        default: True
+        type: bool
+    portal:
+        description:
+            - Red Hat Portal subscription access address
+        default: https://subscription.rhn.redhat.com
+'''
+
+EXAMPLES = '''
+- name: Create katello.example.com Manifest and add 7 sub
+  redhat_manifest:
+    name: "katello.example.com"
+    username: "john-smith"
+    password: "changeme"
+    pool_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    quantity: 7
+    verify_ssl: false
+
+- name: Ensure my manifest has 10 of one subs in it and export
+  redhat_manifest:
+    uuid: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+    username: john-smith
+    password: changeme
+    pool_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    quantity: 10
+    verify_ssl: false
+    path: /root/manifest.zip
+
+- name: Remove all of one subs from katello.example.com
+  redhat_manifest:
+    name: katello.example.com
+    username: john-smith
+    password: changeme
+    pool_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    pool_state: absent
+    verify_ssl: false
+'''
+
+RETURN = '''# '''
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.pycompat24 import get_exception
+
+
+def fetch_portal(module, path, method, data={}, accept_header='application/json'):
+    url = module.params['portal'] + path
+    headers = {'accept': accept_header,
+               'content-type': 'application/json'}
+    resp, info = fetch_url(module, url, json.dumps(data), headers, method)
+    if resp is None:
+        try:
+            error = json.loads(info['body'])['displayMessage']
+        except:
+            error = info['msg']
+        raise Exception(error)
```

Comment:
I have the expression, that this Exception might not be caught (and transformed into module.fail_json) at every invocation of fetch_portal.